### PR TITLE
#168

### PR DESCRIPTION
**Pull request by MJJacobs01**
Good morning,

I have updated Gradle and the dependencies as per #168.

I have also added a schemas output for room to satisfy the compiler.

### CODE REVIEW

**Comment by MJJacobs01** [comment]
@araneforseti 

Please update the Java version from v11 to v17, as that is the reason the tests are failing.

On my side all tests passed

**Comment by araneforseti** [comment]
@MJJacobs01 done. Try rebasing now

**Comment by MJJacobs01** [comment]
May I ask a stupid question.

What is the difference between a new branch and rebasing?

**Comment by araneforseti** [comment]
Rebasing is when you pull updates in. So for instance instead of making a new branch to get the Java update, you can rebase master into your branch. GitHub should show a button to do it automatically too since I set this repo to require branches be up to date.

**Comment by MJJacobs01** [comment]
Thanks.

I will try it.

**Comment by MJJacobs01** [comment]
Good morning,

Please check and let me know if you are happy?

File: app/build.gradle [hunk 1]
```diff
@@ -46,31 +47,33 @@ android {
...
 
     testImplementation 'junit:junit:4.13.2'
     androidTestImplementation 'androidx.test.ext:junit:1.1.5'
     androidTestImplementation 'androidx.test.espresso:espresso-core:3.5.1'
     androidTestImplementation "androidx.arch.core:core-testing:$rootProject.coreTestingVersion"
 
-    kapt 'com.google.dagger:hilt-android-compiler:2.45'
-    implementation 'com.google.dagger:hilt-android:2.45'
+    ksp("com.google.dagger:hilt-compiler:2.50")
+    ksp("androidx.hilt:hilt-compiler:1.1.0")
```
Comment (araneforseti) [inline app/build.gradle:67]: Nitpick here, but please follow the format of the rest of the file and remove the '(' (I need to see if I can get the linter to catch this in the future)

File: app/build.gradle [hunk 2]
```diff
@@ -46,31 +47,33 @@ android {
...
+//    kapt 'com.google.dagger:hilt-android-compiler:2.50'
```
Comment (araneforseti) [inline app/build.gradle:68]: If this line is no longer needed, please delete

File: app/build.gradle [hunk 3]
```diff
@@ -46,31 +47,33 @@ android {
...
+    implementation 'com.google.dagger:hilt-android:2.50'
 
     implementation "androidx.room:room-runtime:$room_version"
     implementation "androidx.room:room-paging:$room_version"
     annotationProcessor "androidx.room:room-compiler:$room_version"
 
-    // To use Kotlin annotation processing tool (kapt)
-    kapt("androidx.room:room-compiler:$room_version")
+    // KSP
+    ksp("androidx.room:room-compiler:$room_version")
```
Comment (araneforseti) [inline app/build.gradle:76]: See above with '('

File: .idea/.gitignore [hunk 4]
```diff
@@ -1,3 +1,4 @@
 # Default ignored files
 /shelf/
 /workspace.xml
+/app/schemas
```
Comment (araneforseti) [inline .idea/.gitignore:4]: Question about this - I saw you added a schema file. Will this being in the gitignore interfere with that and what sort of files are you expecting to see in this directory which you want ignored?

**Review by araneforseti — CHANGES_REQUESTED** [review]
A few questions and some nitpicks. Otherwise looks good so once those are answered or addressed this can be merged.

**Comment by araneforseti** [comment]
Also would like to know why you decided to introduce the schema file. Is there a good practice documented somewhere requiring this?

File: .idea/.gitignore (hunk 4 above)
Comment (MJJacobs01) [reply .idea/.gitignore:4]: This will only be the database version json file, to make future auto migrations possible and easier

**Comment by araneforseti** [comment]
Accepted

**Pull request merged**